tools/ReleaseNotes/release_editor.py

```diff
@@ -0,0 +1,129 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import os
+import sys
+import shutil
+
+DIRECTIVE = ".. amalgamate:: "
+
+
+def getReleaseRoot():
+    script_dir = os.path.split(sys.argv[0])[0]
+    return os.path.abspath(os.path.join(script_dir, '../../docs/source/release/'))
+
+
+def fixReleaseName(name):
+    if not name.startswith('v'):
+        name = 'v' + name
+    return name
+
+
+def createFileLocation(releaseNo):
+    release_root = getReleaseRoot()
+    release_root = release_root + '/' + releaseNo
+    return release_root
+
+
+def updateFile(originalFile, replaceText, textToReplace):
+    # opens the upper level release note e.g. diffraction.rst
+    mainRST = open(originalFile, 'r')
+    rstToRead = mainRST.read()
+    mainRST.close()
+
+    # replaces the amalgamate directive with the notes compiled for that heading
+    newdata = rstToRead.replace(textToReplace, replaceText)
+
+    mainRST = open(originalFile, 'w')
+    mainRST.write(newdata)
+    mainRST.close()
+
+
+def createFileName(fileName, pathDirectory, includeStatement):
+    newFileName = fileName.replace(includeStatement, "")
+    newFileName = newFileName.replace("\n", "")
+    newFileName = pathDirectory + '/' + newFileName
+    return os.path.normpath(newFileName)
+
+
+def addReleaseNotesToMain(pathDirectory):
+    # iterates through files in a directory
+    for file in os.listdir(pathDirectory):
```

Comment:
May I suggest using the [glob](https://docs.python.org/3.8/library/glob.html) module here? It would mean you only get a list of `.rst` files back and don't have to check that yourself: `for filename in glob.glob(pathDirectory.resolve('*.rst')):`